tests/tracer/test_single_span_sampling_rules.py

```diff
@@ -119,6 +119,169 @@ def test_single_span_rules_do_not_tag_if_tracer_samples():
         assert_sampling_decision_tags(span, sample_rate=None, mechanism=None, limit=None, trace_sampling=True)
 
 
+def test_sampling_rule_init_sngl_rule_via_file(tmpdir):
+    file = tmpdir.mkdir("data").join("rules.json")
```

Comment:
you probably don't need `.mkdir("data")` here, `tmpdir` should be some random temporary directory so you can be safe to just do `tmpdir.join("rules.json")`

not that big of a deal, more FYI.